src/workspaces/common/state/useWorkspaceStatePolling.ts

```diff
@@ -99,32 +102,36 @@ export const useWorkspaceStatePolling = (workspaces: Workspace[], status: Loaded
   }, [workspaces, status]); // adding the controller to deps causes a double fire of the effect
 };
 
-// we need a separate implementation of this, because using useWorkspaceStatePolling with a list
-// containing a single workspace in the WorkspaceContainer causes a rendering loop
-export const useSingleWorkspaceStatePolling = (workspace: Workspace) => {
-  const [controller, setController] = useState(new window.AbortController());
+export type StateUpdateAction = (WorkspaceUpdate) => void;
+export type StateUpdateListener = { [key in WorkspaceState]?: StateUpdateAction[] };
+
+// Monitors the state of workspaces and calls the listeners when the state changes
+export const useWorkspacesStatePollingWithAction = (workspaces: WorkspaceInfo[], listeners: StateUpdateListener) => {
+  // we have to do the signal/abort manually instead of with useCancelable so that the it can be cleaned up in the
+  // this component's useEffect, instead of the useEffect in useCancelable
+  const controller = useRef(new window.AbortController());
   const abort = () => {
-    controller.abort();
-    setController(new window.AbortController());
+    controller.current.abort();
+    controller.current = new window.AbortController();
   };
 
   useEffect(() => {
-    pollWithCancellation(
-      async () => {
-        if (_.contains(workspace?.workspace?.state, updatingStates)) {
-          const updatedState = await checkWorkspaceState(workspace, controller.signal);
-          if (updatedState) {
-            abort();
-          }
+    const iterateUpdatingWorkspaces = async () => {
+      // wait for the workspaces to be loaded (LoadedState.status) before polling
+      const updatingWorkspaces = _.filter((ws) => _.contains(ws.state, updatingStates), workspaces);
+      for (const ws of updatingWorkspaces) {
+        const updatedState = await checkWorkspaceState(ws, controller.current.signal);
+        if (updatedState) {
+          const updatedWS = _.merge(ws, updatedState);
+          listeners[updatedState.state]?.forEach((callback) => callback(updatedWS));
+          abort();
         }
-      },
-      30000,
-      false,
-      controller.signal
-    );
+      }
+    };
+
+    pollWithCancellation(() => iterateUpdatingWorkspaces(), 30000, false, controller.current.signal);
     return () => {
       abort();
     };
-    //  eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [workspace]); // adding the controller to deps causes a double fire of the effect
+  }, [workspaces, listeners]); // adding the controller to deps causes a double fire of the effect
```

Comment:
Is the comment still necessary? The `eslint-disable-next-line react-hooks/exhaustive-deps` is no longer present.